draft-ietf-lamps-cms-kyber.md

```diff
@@ -27,474 +38,358 @@ author:
 
-4. Concatenate the wrapped keying data *WK* of length *wrappedKekLen* and the ciphertext *CT* to obtain the encrypted keying data *EK*:
+## RecipientInfo Conventions {#sec-using-recipientInfo}
 
-   > EK = (WK \|\| CT)
+When the ML-KEM Algorithm is employed for a recipient, the RecipientInfo alternative for that recipient MUST be OtherRecipientInfo using the KEMRecipientInfo structure as defined in {{!I-D.ietf-lamps-cms-kemri}}.
+The fields of the KEMRecipientInfo MUST have the following values:
 
-5. Output the encrypted keying data *EK*.
+> version is the syntax version number; it MUST be 0.
 
-<!-- End of sender-operations section -->
+> rid identifies the recipient's certificate or public key.
 
+> kem identifies the KEM algorithm ; For ML-KEM-512 it MUST contain id-ML-KEM-512, for ML-KEM-768 it MUST contain id-ML-KEM-768, for ML-KEM-1024 it MUST contain id-ML-KEM-1024. These identifiers are reproduced in {{sec-identifiers}}.
 
-## Recipient's Operations {#sec-recipient-operations}
+> kemct is the ciphertext produced for this recipient.
 
-This process assumes that the following algorithm parameters have been communicated from the sender:
+> kdf identifies the key-derivation algorithm. Note that the KDF used for CMS RecipientInfo process MAY be different than the KDF used within the ML-KEM Algorithm.
 
-- *KEM*: a key encapsulation mechanism, as defined above.
-- *KDF*: a key derivation function, as defined above.
-- *Wrap*: a symmetric key-wrapping algorithm, as defined above.
-- *kekLen*: the length in bits of the key required for the Wrap algorithm.
+> kekLength is the size of the key-encryption key in octets.
 
-This process assumes that the following input data has been provided:
+> ukm is an optional random input to the key-derivation function. ML-KEM doesn't place any requirements on the ukm contents.
 
-- *recipPrivKey*: the recipient's private key.
-- *EK*: the encrypted keying data.
+> wrap identifies a key wrapping algorithm used to encrypt the content-encryption key.
 
-This process outputs:
+<!-- End of recipientinfo conventions section -->
 
-- *K*: the keying data to be transported.
+## Underlying Components {#sec-using-components}
 
-The recipient performs the following operations:
+When ML-KEM is employed in CMS, the security levels of the different underlying components used within the KEMRecipientInfo structure should be consistent.
 
-1. Separate the encrypted keying data *EK* into wrapped keying data *WK* of length *wrappedKekLen* and a ciphertext *CT* :
+\[EDNOTE: if we get OIDs for KMAC-based KDFs, use those. If we don't, do we want to use KDF3 with SHA3 <!--{{!ANS-X9.44=ANSI.X9-44.1993}}--> instead? ]
```

Comment:
Yup! I agree with this ednote; this section will be in flux (and I will need to align draft-composite-kem with whatever choice is made here).

Reply:
BTW I heard directly from Russ that we'll get OIDs for KMAC-based KDFs.  If NIST doesn't assign them as he asked then LAMPS will.